dpe/src/dpe_instance.rs

```diff
@@ -37,6 +40,17 @@ impl<C: Crypto> DpeInstance<'_, C> {
     const MAX_NEW_HANDLE_ATTEMPTS: usize = 8;
     pub const AUTO_INIT_LOCALITY: u32 = 0;
 
+    pub(crate) fn new_context_handles() -> [Context<C>; MAX_HANDLES] {
+        unsafe {
+            let mut array: MaybeUninit<[Context<C>; MAX_HANDLES]> = MaybeUninit::uninit();
+            for i in 0..MAX_HANDLES {
+                let arr = &mut *array.as_mut_ptr();
+                arr[i] = Context::new();
+            }
+            array.assume_init()
+        }
```

Comment:
I don't think I understand why this part is necessary. It seems like `const CONTEXT_INITIALIZER: Context<C> = Context<C>::new();` should be sufficient.

Reply:
Indeed, I'm not sure I'm following either. I tried the following is the tests in `src/context.rs` and it worked:

```rust
        const INITIALIZER_CONTEXT: Context<OpensslCrypto> = Context::new();
        let mut contexts = [INITIALIZER_CONTEXT; MAX_HANDLES];
```

Reply:
It looks like the problem is the generic in a `const`. You can do it if the type is known, but not if it is defined yet (i.e. `C` instead of `OpensslCrypto`). I got it to work with this:

```Rust
    pub(crate) fn new_context_handles() -> [Context<C>; MAX_HANDLES] {
        [(); MAX_HANDLES].map(|_| Context::<C>::new())
    }
```

The `map` function says its not optimized very well, but that shouldn't matter because it only happens once on initialization.

Reply:
There's also `core::array::from_fn(|_| Context::new())` which is probably a bit more straightforward. Here's a patch: https://github.com/sree-revoori1/caliptra-dpe/compare/main...ia0:caliptra-dpe:pr89. A helper function `new_context_handles` may still be used though.